src/fhir_search/uri_parser.clj

```diff
@@ -0,0 +1,140 @@
+(ns fhir-search.uri-parser
+  (:require [clojure.string :as str])
+  (:require [clojure.walk :refer [postwalk]])
+  (:import [java.net URI]))
+
+(def prefixes ["eq" "ne" "gt" "lt" "ge" "le" "sa" "eb" "ap"])
+
+(defn prefix-find "Find the prefix of a number value. If there are more than two prefixes or the value is not the correct returns nil."
+  [pf-list element]
+  (when-let [prefix (first (filter (and #(re-find (re-pattern %) element)
+                                        #(not (re-find #"\p{L}" (str/replace element (re-pattern %) "")))) pf-list))]
+    prefix))
+
+
+(defn url-encoding [url]
+  (let [uri (URI. url)
+        scheme (.getScheme uri)
+        authority (.getAuthority uri)
+        path (.getPath uri)
+        query (.getQuery uri)
+        fragment (.getFragment uri)]
+    (when uri
+      (str (when scheme (str scheme "://"))
+           authority path
+           (when query (str "?" query))
+           (when fragment (str "#" fragment))))))
+
+(defn values [query]
```

Comment:
It is easy to see that this fn has two responsibilities, parsing the `first-group` and parsing the `second-group`, where the first first-group is the parameter's name and the second-group is the parameter's value. Can we split up this fn into `(param-name` and `(param-value`? Each one has it's own set of parsing rules. For example, the name can contain mofiers, or represent a chain, and the value can contain prefixes, comma separated values, etc.